### ISSUE TITLE
Add autocomplete to GAMA Platform

### ISSUE THREAD

**Issue by mikhailsirenko** [gama-platform/gama#3007]
Similary to many IDEs it would be cool to have an option to turn on [autocomplete](https://en.wikipedia.org/wiki/Autocomplete#:~:text=Autocomplete%2C%20or%20word%20completion%2C%20is,to%20accept%20one%20of%20several.). I believe it can significantly speed up/simplify the coding process. Perhaps as a starting point just add autcomplete for user-defined variables? The next step could be something more elaborate, e.g. when a user can see the methods, extra documentation etc.

**Comment by benoitgaudou** [comment]
Dear,

In fact, a form of autocompletion  exists in GAMA, in fact very close  to
the one we can have on Eclipse and for Java.
I check a few minutes (perhaps I missed  some  features...),  but I
identify 2 kinds of auto-completion (note that if  the popup does not
appear, you can open it with Ctr+SPACE (on Mac ... if  this shortcut  is
not used  for something else):
 - attributes of  agents: when you have  an agent (e.g. self), and  want to
access an attribute, after writing the  ., some possibilities  appear:
[image: Screenshot 2020-07-02 at 10.52.58.png]

-  a second option is when you start a line, ctrl+space can propose you
available statements. Note that it also proposes access to templates.
See below when I write "loo", and  ctrl+Space :

[image: Screenshot 2020-07-02 at 11.11.17.png]


To sum up, some autocompletion is available on GAMA.
If you have more precise ideas in mind, do not hesitate to let us know.

Cheers

Benoit




Le jeu. 2 juil. 2020 à 01:13, Mikhail Sirenko <notifications@github.com> a
écrit :

**Comment by mikhailsirenko** [comment]
Hi benoitgaudou!

That's amazing! I was struggling to find this feature by searching for "autcomplete"/"auto-complete" in the docs. The combination CTRL+Space is unfamiliar to me (I switched from Python/R JupyterLab/VS Code), but now I know :-) Again, thanks a lot!
